scripts/component-sizes.js

```diff
@@ -8,355 +8,99 @@ LICENSE file in the root directory of this source tree.
-  .BundleAnalyzerPlugin;
-
-function modulePath(identifier) {
-  // the format of module paths is
-  //   '(<loader expression>!)?/path/to/module.js'
-  let loaderRegex = /.*!/;
-  return identifier.replace(loaderRegex, '');
-}
-
-function getByteLen(normal_val) {
-  // Force string type
-  normal_val = String(normal_val);
-
-  var byteLen = 0;
-  for (var i = 0; i < normal_val.length; i++) {
-    var c = normal_val.charCodeAt(i);
-    byteLen +=
-      c < 1 << 7
-        ? 1
-        : c < 1 << 11
-        ? 2
-        : c < 1 << 16
-        ? 3
-        : c < 1 << 21
-        ? 4
-        : c < 1 << 26
-        ? 5
-        : c < 1 << 31
-        ? 6
-        : Number.NaN;
-  }
-  return byteLen;
-}
-
-function minify(source) {
-  return Terser.minify(source, {
-    mangle: false,
-    compress: {
-      arrows: true,
-      booleans: true,
-      collapse_vars: true,
-      comparisons: true,
-      conditionals: true,
-      dead_code: true,
-      drop_console: false,
-      drop_debugger: true,
-      ecma: 5,
-      evaluate: true,
-      expression: false,
-      global_defs: {},
-      hoist_vars: false,
-      ie8: false,
-      if_return: true,
-      inline: true,
-      join_vars: true,
-      keep_fargs: true,
-      keep_fnames: false,
-      keep_infinity: false,
-      loops: true,
-      negate_iife: true,
-      passes: 1,
-      properties: true,
-      pure_getters: 'strict',
-      pure_funcs: null,
-      reduce_vars: true,
-      sequences: true,
-      side_effects: true,
-      switches: true,
-      top_retain: null,
-      toplevel: false,
-      typeofs: true,
-      unsafe: false,
-      unused: true,
-      warnings: false,
-    },
-    output: {
-      comments: false,
-    },
-  });
-}
-
-function getDependencySizes(stats) {
-  let statsTree = {
-    packageName: '<root>',
-    sources: [],
-    children: [],
-  };
-
-  if (stats.name) {
-    statsTree.bundleName = stats.name;
-  }
-
-  if (!stats.modules) return [];
+main().catch(console.error);
 
-  // extract source path for each module
-  let modules = [];
-  const makeModule = mod => {
-    // Uglifier cannot minify a json file, hence we need
-    // to make it valid javascript syntax
-    const isJSON = mod.identifier.endsWith('.json');
-    const source = isJSON ? `$a$=${mod.source}` : mod.source;
+async function main() {
+  const original = require('../component-sizes.json');
 
-    return {
-      path: modulePath(mod.identifier),
-      sources: [source],
-      source: source,
-    };
+  const componentExports = {
+    accordion: ['Accordion', 'Panel'],
+    avatar: 'Avatar',
+    block: 'Block',
+    breadcrumbs: 'Breadcrumbs',
+    'button-group': 'ButtonGroup',
+    button: 'Button',
+    card: 'Card',
+    checkbox: 'Checkbox',
+    'data-table': 'Unstable_StatefulDataTable',
+    datepicker: 'Datepicker',
+    'dnd-list': 'List',
+    drawer: 'Drawer',
+    'file-uploader': 'FileUploader',
+    'flex-grid': 'FlexGrid',
```

Comment:
```suggestion
    'flex-grid': 'FlexGrid',
    'form-control': 'FormControl',
```

Reply:
It's a bit unfortunate that this manually created map is required. Leads to missing data like above and adds another requirement on top of the many things necessary to contribute a new component. I suppose the problem is difficult though, since you need to select an array of exports in some situations

Reply:
Yeah, I agree, and I am not sure how to better automate it for now :/